gateway/ln-gateway/src/lightning/lnd.rs

```diff
@@ -732,6 +732,57 @@ impl ILnRpcClient for GatewayLndClient {
         }
     }
 
+    async fn close_channels_with_peer(&self, pubkey: PublicKey) -> Result<u32, LightningRpcError> {
+        let mut client = self.connect().await?;
+
+        let channels_with_peer = client
+            .lightning()
+            .list_channels(ListChannelsRequest {
+                active_only: false,
+                inactive_only: false,
+                public_only: false,
+                private_only: false,
+                peer: pubkey.serialize().to_vec(),
+            })
+            .await
+            .map_err(|e| LightningRpcError::FailedToCloseChannelsWithPeer {
+                failure_reason: format!("Failed to list channels {e:?}"),
+            })?
+            .into_inner()
+            .channels;
+
+        for channel in &channels_with_peer {
```

Comment:
A failure in this loop would be annoying since the first channels are already closed while later ones stay open. Seems like a potential source of bugs. I'd much prefer the "single channel closing" semantics eventually.

Reply:
This function doesn't wait for the channel to close anyway, we'll eventually need a way to track that as well. Rather than failing and returning here (and above in the CLN implementation), I think emitting a warning and trying to close the next channel would be sufficient